File: App/ios-tms/Models/Shipment.swift
//
//  Shipment.swift
//  ios-tms
//
//  Created by Kyle Morton on 8/28/22.
//

import Foundation

struct Shipment: Codable, Identifiable {
    
    // simplifing model to the below (for now)
    var id: Int
    var bol: String
    var origin: String
    var destination: String
    var carrier: String
    var items: Int
    var weight: Decimal
    var rate: Decimal
    
    init(id: Int, bol: String, origin: String, destination: String, carrier: String, items: Int, weight: Decimal, rate: Decimal) {
        self.id = id
        self.bol = bol
        self.origin = origin
        self.destination = destination
        self.carrier = carrier
        self.items = items
        self.weight = weight
        self.rate = rate
    }
    
//    var originId: Int?
//    var destinationId: Int?
//    var carrierId: Int?
//    var origin: Address?
//    var destination: Address?
//    var carrier: Carrier?
//    var items: [ShipmentItem]
//    var rate: Decimal?
//    var cost: Decimal?
    
//    init(id: Int, origin: Address, destination: Address, carrier: Carrier, items: [ShipmentItem], customerRate: Decimal) {
//        self.id = id
//        self.origin = origin
//        self.destination = destination
//        self.carrier = carrier
//        self.items = items
//        self.rate = customerRate
//    }
    
    
    
}

File: API/ios-tms.Core/Domain/Shipment.cs
using System;
namespace ios_tms.Core.Domain;

public class Shipment : EntityBase
{
    public string Bol { get; set; }
    public string Origin { get; set; }
    public string Destination { get; set; }
    public string Carrier { get; set; }
    public int Items { get; set; }
    public Decimal Weight { get; set; }
    public Decimal Rate { get; set; }
    //public int OriginId { get; set; }
    //public int DestinationId { get; set; }
    //public int CarrierId { get; set; }
    //public Decimal Rate { get; set; }
    //public Decimal Cost { get; set; }
    //public virtual Address Origin { get; set; }
    //public virtual Address Destination { get; set; }
    //public virtual Carrier Carrier { get; set; }
    //public virtual List<ShipmentItem> Items { get; set; }
}



